db/user.go

```diff
@@ -221,3 +222,72 @@ func SyncUsers(course *model.Course, newUsers map[string]*model.User,
 
     return syncResult, nil;
 }
+
+// Convert roles to the associated user emails for a given course.
+// Returns a slice of lowercase emails without duplicates.
+// Will error if given non-emails without a valid course.
+// Will warn if given roles that do not exist within a course.
```

Comment:
"Roles within a course" does not make sense.
All courses have the same set of available roles.